membrane_curvature/surface.py

```diff
@@ -131,3 +131,60 @@ def normalized_grid(grid_z_coordinates, grid_norm_unit):
     z_normalized = grid_z_coordinates / grid_norm_unit
 
     return z_normalized
+
+
+def interpolation_by_array(array_surface):
+    """
+        Interpolates values contained in `array_surface` over axis
+
+    Parameters
+    ----------
+
+    array_surface: np.ndarray
+        Array of floats of shape (`n_x_bins`, `n_y_bins`)
+
+    Returns
+    -------
+    interpolated_array: np.ndarray
+        Returns interpolated array.
+        Array of shape (`n_x_bins` x `n_y_bins`,)
+
+    """
+
+    # create mask for nans
+    mask_nans = ~np.isnan(array_surface)
+
+    # index of array_surface
+    index_array = np.arange(array_surface.shape[0])
+
+    # interpolate values in array
+    interpolated_array = np.interp(index_array,
+                                   np.flatnonzero(mask_nans),
+                                   array_surface[mask_nans])
+
+    return interpolated_array
```

Comment:
I'm not quite understanding this. `mask_nans` is 2D, with shape (n_x, n_y). `index_array` is 1D, with shape `n_x`. `np.flatnonzero(mask_nans)` and `array_surface[mask_nans]` will be 1D, of length between 0 to n_x * n_y. So I *think* that means that you are only ever operating on the first row of `array_surface`?

I guess my question is then, why is `array_surface` 2D? Why not just pass in one row at a time? In addition, as it is a 2D surface, why use a 1D interpolation function? Why not a [2D one](https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.interp2d.html)?